Analysis/DataModel/include/AnalysisDataModel/HFSecondaryVertex.h

```diff
@@ -311,6 +318,100 @@ DECLARE_SOA_TABLE(HfCandProng2MCGen, "AOD", "HFCANDP2MCGEN",
                   hf_cand_prong2::FlagMCMatchGen,
                   hf_cand_prong2::OriginMCGen);
 
+// cascade decay candidate table
+
+namespace hf_cand_casc
+{
+DECLARE_SOA_EXPRESSION_COLUMN(Px, px, float, 1.f * aod::hf_cand::pxProng0 + 1.f * aod::hf_cand::pxProng1);
+DECLARE_SOA_EXPRESSION_COLUMN(Py, py, float, 1.f * aod::hf_cand::pyProng0 + 1.f * aod::hf_cand::pyProng1);
+DECLARE_SOA_EXPRESSION_COLUMN(Pz, pz, float, 1.f * aod::hf_cand::pzProng0 + 1.f * aod::hf_cand::pzProng1);
+//DECLARE_SOA_DYNAMIC_COLUMN(M, m, [](float px0, float py0, float pz0, float px1, float py1, float pz1, const array<double, 2>& m) { return RecoDecay::M(array{array{px0, py0, pz0}, array{px1, py1, pz1}}, m); });
+DECLARE_SOA_DYNAMIC_COLUMN(PtV0Pos, ptV0Pos, [](float px, float py) { return RecoDecay::Pt(px, py); });
+DECLARE_SOA_DYNAMIC_COLUMN(PtV0Neg, ptV0Neg, [](float px, float py) { return RecoDecay::Pt(px, py); });
+DECLARE_SOA_COLUMN(FlagMCMatchRec, flagMCMatchRec, int8_t); // reconstruction level
+DECLARE_SOA_COLUMN(FlagMCMatchGen, flagMCMatchGen, int8_t); // generator level
+
+template <typename T>
+auto InvMassLcToK0sP(const T& candidate)
+{
+  return candidate.m(array{RecoDecay::getMassPDG(kK0Short), RecoDecay::getMassPDG(kProton)}); // first daughter is K0s
+}
+
+template <typename T>
+auto InvMassGamma(const T& candidate)
+{
+  return candidate.m(array{RecoDecay::getMassPDG(kElectron), RecoDecay::getMassPDG(kElectron)});
+}
+
+} // namespace hf_cand_casc
+
+DECLARE_SOA_TABLE(HfCandCascBase, "AOD", "HFCANDCASCBASE",
+                  // general columns
+                  HFCAND_COLUMNS,
+                  // cascade specific columns
+                  hf_cand::PxProng0, hf_cand::PyProng0, hf_cand::PzProng0,
+                  hf_cand::PxProng1, hf_cand::PyProng1, hf_cand::PzProng1,
+                  hf_cand::ImpactParameter0, hf_cand::ImpactParameter1,
+                  hf_cand::ErrorImpactParameter0, hf_cand::ErrorImpactParameter1,
+                  hf_track_index::Index0Id,
+                  hf_track_index::IndexV0Id, // V0 index
+                  hf_track_index::HFflag,
+                  // V0
+                  v0data::X, v0data::Y, v0data::Z,
+                  v0data::PosTrackId, v0data::NegTrackId, // indices of V0 tracks in FullTracks table
+                  v0data::PxPos, v0data::PyPos, v0data::PzPos, v0data::PxNeg, v0data::PyNeg, v0data::PzNeg,
+                  v0data::DCAV0Daughters,
+                  v0data::DCAPosToPV, // this is the impact param wrt prim vtx in xy!
+                  v0data::DCANegToPV, // this is the impact param wrt prim vtx in xy!
```

Comment:
Are these not accessible from the V0 candidate?

Reply:
Yes, but don't we want them in our table so that it suffices for analysis?